docs/narr/viewconfig.rst

```diff
@@ -1028,6 +1029,76 @@ these values.
 .. index::
    single: HTTP caching
 
+.. _accept_content_negotation:
+
+Accept Header Content Negotiation
+---------------------------------
+
+The ``accept`` argument to :meth:`pyramid.config.Configurator.add_view` can be used to control :term:`view lookup` by dispatching to different views based on the HTTP ``Accept`` request header.
+Consider the example below in which there are three defined views.
+Each view uses the ``Accept`` header to trigger an appropriate response renderer.
+
+.. code-block:: python
+
+    from pyramid.httpexceptions import HTTPNotAcceptable
+    from pyramid.view import view_config
+
+    @view_config(accept='application/json', renderer='json')
+    @view_config(accept='text/html', renderer='templates/hello.jinja2')
+    def myview(request):
+        return {
+            'name': request.GET.get('name', 'bob'),
+        }
+
+    @view_config()
+    def myview_unacceptable(request):
+        raise HTTPNotAcceptable
+
+The appropriate view is selected here when the client specifies an unambiguous header such as ``Accept: text/*`` or ``Accept: application/json``.
+Similarly, if the client specifies a media type that no view is registered to handle, such as ``Accept: text/plain``, it will fall through to ``myview_unacceptable`` and raise ``406 Not Acceptable``.
+There are a few cases in which the client may specify ambiguous constraints:
+
+- ``Accept: */*``.
+- A missing ``Accept`` header.
+- An invalid ``Accept`` header.
+
+In these cases the preferred view is not clearly defined (see :rfc:`7231#section-5.3.2`) and :app:`Pyramid` will select one randomly.
+This can be controlled by telling :app:`Pyramid` what the preferred relative ordering is between various media types by using :meth:`pyramid.config.Configurator.add_accept_view_order`.
+For example:
+
+.. code-block:: python
+
+    from pyramid.config import Configurator
+
+    def main(global_config, **settings):
+        config = Configurator(settings=settings)
+        config.add_accept_view_order('text/html')
+        config.add_accept_view_order(
+            'application/json',
+            weighs_more_than='text/html',
+        )
+        config.scan()
+        return config.make_wsgi_app()
+
+In this case, the ``application/json`` view should always be selected in cases where it is otherwise ambiguous.
+
+Default Accept Ordering
```

Comment:
Insert ``.. _default-accept-ordering:`` for indexing.

Reply:
I merged two thoughts together in my original comment. One is the reference which gets converted to an id and may be linked to, the other is adding an index entry for new sections. Sorry about that.

Sphinx will generate ids for all levels of section headings, whether or not we add a reference. It's not required, but we can control it. But more importantly, without a reference, you can't do this in reST:

```
:ref:`default-accept-ordering`
```

...and instead hyperlinks would have to be grabbed from the rendered HTML and inserted into the reST. That's always bothered me, especially when using intersphinx, so I tend to add references for every section. You never know who will link back to a given section.

As far as index entries, that's up to you if you think they're useful here. I think we've lately been slacking off on them, and I haven't been paying much attention to them. I guess a good rule of thumb is to ask oneself, "Would index entries help make this section easy to find?"

Reply:
I've never once used the index directive and I've never deep linked to subsections. At the very least I don't really want to talk about any of this until I can get some actual code review on the substance of this PR because I have already re-written the docs section multiple times. If you want me to add some of these directives for the index then please paste examples because I just don't have much motivation for them. Sorry :-(

Reply:
Please let me know when you want me to review docs again.

Reply:
@stevepiercy please review the docs again when you get a chance